packages/neutrino/test/api_test.js

```diff
@@ -133,135 +133,53 @@ test('override options.mains', t => {
-  t.is(typeof Neutrino().config.toConfig, 'function');
+  t.is(typeof new Neutrino().config.toConfig, 'function');
 });
 
 test('middleware receives API instance', t => {
-  const api = Neutrino();
+  const api = new Neutrino();
 
   api.use(n => t.is(n, api));
 });
 
 test('middleware receives no default options', t => {
-  const api = Neutrino();
+  const api = new Neutrino();
 
   api.use((api, options) => {
     t.is(options, undefined);
   });
 });
 
 test('middleware receives options parameter', t => {
-  const api = Neutrino();
+  const api = new Neutrino();
   const defaults = { alpha: 'a', beta: 'b', gamma: 'c' };
 
   api.use((api, options) => {
     t.deepEqual(options, defaults);
   }, defaults);
 });
 
-test('triggers promisified event handlers', t => {
-  const api = Neutrino();
-
-  api.on('test', () => t.pass('test event triggered'));
-  api.emitForAll('test');
-});
-
-test('events handle promise resolution', async t => {
-  const api = Neutrino();
-
-  api.on('test', () => Promise.resolve('alpha'));
-
-  const [value] = await api.emitForAll('test');
-
-  t.is(value, 'alpha');
-});
-
-test('events handle promise rejection', async t => {
-  const api = Neutrino();
-
-  api.on('test', () => Promise.reject(new Error('beta')));
-
-  const err = await t.throws(api.emitForAll('test'));
-
-  t.is(err.message, 'beta');
-});
-
-test('events handle multiple promise resolutions', async t => {
-  const api = Neutrino();
-
-  api.on('test', () => Promise.resolve('alpha'));
-  api.on('test', () => Promise.resolve('beta'));
-  api.on('test', () => Promise.resolve('gamma'));
-
-  const values = await api.emitForAll('test');
-
-  t.deepEqual(values, ['alpha', 'beta', 'gamma']);
-});
-
 test('import middleware for use', async (t) => {
-  const api = Neutrino({ root: __dirname });
+  const api = new Neutrino({ root: __dirname });
 
   api.use(['fixtures/middleware']);
   t.notDeepEqual(api.config.toConfig(), {});
 });
 
-test('command emits events around execution', async (t) => {
-  const api = Neutrino();
-  const events = [];
-
-  api.on('prebuild', () => events.push('alpha'));
-  api.on('build', () => events.push('beta'));
-
-  await api.emitForAll('prebuild');
-  await api.emitForAll('build');
-
-  t.deepEqual(events, ['alpha', 'beta']);
-});
-
-test('sets environment variables from options', t => {
-  Neutrino({
-    env: { NODE_ENV: 'production', ALPHA: 'beta' }
-  });
-
-  t.is(process.env.NODE_ENV, 'production');
-  t.is(process.env.ALPHA, 'beta');
-});
-
 test('creates a webpack config', t => {
-  const api = Neutrino();
+  const api = new Neutrino();
 
-  api.use(api => api.config.module
-    .rule('compile')
-    .test(api.regexFromExtensions(['js'])));
+  api.use(api => {
+    api.config.module
+      .rule('compile')
+      .test(api.regexFromExtensions(['js']))
+  });
 
   t.notDeepEqual(api.config.toConfig(), {});
 });
 
-test('throws when trying to call() a non-registered command', t => {
```

Comment:
Do you think we should turn this into a test checking for the `Unable to find an output handler named` behaviour?

Reply:
Added this to the new `package_test`.